### PR TITLE
Re-structuring of server and removal of broken react component (main)

### PR DESCRIPTION
**Pull request by gregtuc**
The server index.js now uses express's Router to modularize how routes are accessed. Routes are now kept in the "routes" folder. For the database, an instance of the database can be called by declaring const client = require(path-to-connect.js). This instance also gets user details now from a default.json file so that we no longer have to continuously add and remove our user details. For creating your own routes, you can refer to the loginUser.js/registerUser.js/searchUser.js files as their template can be followed.

### CODE REVIEW

File: server/database/default.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "ConnectionUrl": "mongodb+srv://greg34910:cookie123@cluster0-pf0b2.mongodb.net/test?retryWrites=true&w=majority"
```
Comment (ginywiny) [inline server/database/default.json]: This is using your own personal DB connection. Can you make this generic?

Comment (gregtuc) [reply server/database/default.json]: Hidden

File: server/routes/loginUser.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      if (!user) {
+        console.log("404");
+        return false;
+      } else {
+        console.log("200");
+        return true;
+      }
+    });
+}
+
```
Comment (ginywiny) [inline server/routes/loginUser.js:28]: Would you be able to add comments around these routing sections, so it is easier to read when looking back?

**Comment by ginywiny** [comment]
I notice that you are routing to different pages. However, would it be possible to display this information to the user (such as through the URL), so they know which page they are on?

File: server/routes/loginUser.js (hunk 2 above)
Comment (gregtuc) [reply server/routes/loginUser.js:28]: Commented my code, except for login.js and signup.js as they are mostly standard react components

Comment (gregtuc) [reply server/routes/loginUser.js:28]: > I notice that you are routing to different pages. However, would it be possible to display this information to the user (such as through the URL), so they know which page they are on?

Yeah I believe that functionality will come in afterwards, as currently the app is only at the point where we render the pages we want to test at root (just http://localhost:8000/). Once we get there, take a look at React's browser router: https://reacttraining.com/react-router/web/api

**Pull request merged**